File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> set = new HashSet();
        while (n != 1) {
            int sum = 0;
            int current = n;
            while (current > 0) {
                int digit = current % 10;
                sum += digit * digit;
                current /= 10;
            }
            if (set.contains(sum)) return false;
            n = sum;    
            set.add(sum);    
        }
        return true;
    }
}

File: 0202-happy-number/0202-happy-number.kt
class Solution {
    fun isHappy(n: Int): Boolean {
        val set = HashSet<Int>();
        var number = n
        while (number != 1) {
            var sum = 0
            var current = number;
            while (current > 0) {
                val digit = current % 10
                sum += digit * digit
                current /= 10
            }
            if (set.contains(sum)) return false;
            set.add(sum)
            number = sum
        }
        return true;
    }
}